ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/MainnetBlockImporter.java

```diff
@@ -37,6 +37,9 @@ public synchronized boolean importBlock(
       final Block block,
       final HeaderValidationMode headerValidationMode,
       final HeaderValidationMode ommerValidationMode) {
+    if (context.getBlockchain().contains(block.getHash())) {
```

Comment:
What's the underlying `BlockHashFunction` that get's used for IBFT? ...is that hash going to match the hash from the mainnet `BlockHashFunction`, or could they be different due to the IBFT header funniness?

Reply:
This importer is used by only IBFT when we are using IBFT despite the name. It won't be comparing a block hash from mainnet to a block hash from IBFT. The underlying hash function that will be used is different and the hash function that will be used is the IbftBlockHashing::calculateHashOfIbftBlockOnChain which takes cares of header funniness you mentioned.

Reply:
To be clear, that importer is used by MainNet and IBFT, just Pantheon can't be simultaneously doing Ethash and IBFT.  So all blocks will use a consistent block hash function.

By definition, two blocks are equal if and only if their block hash matches. Most important application of that principal is the fact that block headers are stored by their hash in RocksDB.